### DISCUSSION TITLE
How to offer multiple homepage options in a single theme

### DISCUSSION THREAD

**Discussion by annezazu** [WordPress/developer-blog-content#66] [Topic Ideas]
This touches on what are best practices, naming conventions for style variations, and more but I could see this becoming more of a use case, especially since this is part of what's powerfully unlocked with block themes.

**Comment by bph** [comment]
@justintadlock Is this similar to your topic idea? 

- https://github.com/WordPress/developer-blog-content/discussions/63

**Reply by justintadlock** [reply]
Sorry, I missed this when you first asked.  I could see my proposed topic potentially fitting under the umbrella of this one, depending on how fleshed out this idea is.  Mine covered a specific use case, and this topic is a bit general and wide open.

**Comment by DaisyOlsen** [comment]
@annezazu This is the issue that Justin was referring to in #63, for more context. From reading the title and description, this could go a few different directions. My first take was including multiple custom templates with a home page feel in a theme. I don't know that I see the connection to naming conventions for style variations with home page template options. Unless I missed a big shift, variations are applied site-wide and not at the post/page or template level.

**Reply by annezazu** [reply]
Yes! I recognize this was the issue but I'm not quite sure what feedback @justintadlock needs feedback around. In case it helps, I am writing this more from the theme author perspective and helping theme shops build in multiple homepage options. This could be as easy as having different patterns to pick from or something more like having multiple custom templates to choose from. 

I see this as a more umbrella post with https://github.com/WordPress/developer-blog-content/discussions/63 being a specific type of examples that could fit within the larger post, along with using patterns, custom templates, etc. For context, I'm thinking of how to best guide folks to create themes like this https://www.anarieldesign.com/themes/yuna/ that have multiple different layout options for the homepage.

**Reply by DaisyOlsen** [reply]
Gotcha, my thought is that this general post would be about the tools available. Templates, patterns, style variations, and parts and how they can all work together to create options with a mix & match approach. 

That isn't isolated only to home pages but it is a good example to build from.

**Reply by justintadlock** [reply]
> Yes! I recognize this was the issue but I'm not quite sure what feedback @justintadlock needs feedback around. In case it helps, I am writing this more from the theme author perspective and helping theme shops build in multiple homepage options. This could be as easy as having different patterns to pick from or something more like having multiple custom templates to choose from.

The idea around patterns is actually helpful. That might even be a good example for the [templates types for patterns post](https://github.com/WordPress/developer-blog-content/discussions/58).

But, I see #63 as clearly being a separate post.

**Comment by justintadlock** [comment]
So, far, we have one [published post](https://github.com/WordPress/developer-blog-content/issues/81) and one [planned tutorial](https://github.com/WordPress/developer-blog-content/issues/55) on template types for patterns that will dive into this topic. I think these should sufficiently cover some of the ideas from this discussion. It's just so open-ended that I'm not sure if there are more things we should explore here or just open up specific topic ideas for them..

**Comment by bph** [comment]
Published: [Using template patterns to build multiple homepage designs](https://developer.wordpress.org/news/2023/04/using-template-patterns-to-build-multiple-homepage-designs/)

@annezazu does the recently published post cover, what you thought about when you posted the idea?

**Reply by annezazu** [reply]
Yeah! This seems on point. Closing out :)
